File: mac/Sources/Insight/Theme.swift
import AppKit

/// Enough of the website's look that the two feel like one product.
///
/// The windows are forced to dark aqua rather than following the system, so a
/// student who pairs from a dark page doesn't then meet a bright grey dialog
/// that looks like it came from somewhere else.
enum Theme {
    static let background = NSColor(srgbRed: 0.055, green: 0.059, blue: 0.067, alpha: 1)
    static let surface = NSColor(srgbRed: 0.086, green: 0.094, blue: 0.106, alpha: 1)
    static let accent = NSColor(srgbRed: 0.357, green: 0.659, blue: 0.961, alpha: 1)
    static let good = NSColor(srgbRed: 0.290, green: 0.765, blue: 0.541, alpha: 1)
    static let bad = NSColor(srgbRed: 0.898, green: 0.478, blue: 0.478, alpha: 1)

    static let body = NSFont.systemFont(ofSize: 13)
    static let heading = NSFont.systemFont(ofSize: 17, weight: .semibold)
    static let small = NSFont.systemFont(ofSize: 11)

    /// Text colours come from the system so they stay legible if Apple changes
    /// what dark mode means.
    static let text = NSColor.labelColor
    static let textMuted = NSColor.secondaryLabelColor
    static let textFaint = NSColor.tertiaryLabelColor
}

File: windows/Theme.cs
using System.Drawing;

namespace Insight;

/// <summary>
/// Enough of the website's look that the two feel like one product.
///
/// A student pairs this from a dark page and then meets a grey Windows dialog;
/// matching the surface, line and text colours is most of what stops that
/// feeling like a different, less trustworthy program.
/// </summary>
internal static class Theme
{
    internal static readonly Color Background = Color.FromArgb(0x13, 0x13, 0x16);
    internal static readonly Color Surface = Color.FromArgb(0x1B, 0x1B, 0x20);
    internal static readonly Color Line = Color.FromArgb(0x26, 0x26, 0x2C);
    internal static readonly Color Text = Color.FromArgb(0xED, 0xED, 0xF0);
    internal static readonly Color TextMuted = Color.FromArgb(0x9C, 0x9C, 0xA8);
    internal static readonly Color TextFaint = Color.FromArgb(0x7A, 0x7A, 0x87);
    internal static readonly Color Accent = Color.FromArgb(0x1F, 0xA9, 0x7A);
    internal static readonly Color Good = Color.FromArgb(0x3F, 0xBF, 0x8F);
    internal static readonly Color Bad = Color.FromArgb(0xE2, 0x56, 0x3F);

    internal static readonly Font Body = new("Segoe UI", 10F);
    internal static readonly Font Heading = new("Segoe UI Semibold", 13F);
    internal static readonly Font Small = new("Segoe UI", 8.5F);
}
